packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.js

```diff
@@ -42,68 +57,69 @@ describe('withInfiniteScroll', () => {
   };
 
   const InfiniteList = withInfiniteScroll(List);
-  let intersect;
-
-  beforeEach(() => {
-    global.IntersectionObserver = class {
-      constructor(callback) {
-        intersect = async () => callback([{ intersectionRatio: 1 }]);
-      }
-      observe() {} // eslint-disable-line
-      unobserve() {} // eslint-disable-line
-    };
-  });
 
   it('renders an empty list for the first render', () => {
     const tree = renderer.create(
-      <InfiniteList onItemsFetch={onItemsFetch(elementsArray)} />,
+      <InfiniteList dataSource={new MyDS(onItemsFetch(elementsArray))} />,
     );
 
     expect(tree).toMatchSnapshot();
   });
 
-  it('renders items after the first render', async () => {
+  it('renders items after the first render', done => {
     const tree = mount(
-      <InfiniteList onItemsFetch={onItemsFetch(elementsArray)} />,
+      <InfiniteList dataSource={new MyDS(onItemsFetch(elementsArray))} />,
     );
 
-    await intersect();
-    tree.update();
+    setTimeout(async () => {
+      tree.update();
+      await mockIntersect();
+      tree.update();
 
-    expect(toJson(tree)).toMatchSnapshot();
+      expect(toJson(tree)).toMatchSnapshot();
+      done();
+    }, 1);
   });
 
-  it('should render correctly when no more elements', async () => {
+  it('should render correctly when no more elements', done => {
     const tree = mount(
-      <InfiniteList onItemsFetch={onItemsFetch(elementsArray)} />,
+      <InfiniteList dataSource={new MyDS(onItemsFetch(elementsArray))} />,
     );
 
-    await intersect();
-    tree.update();
-    await intersect();
-    tree.update();
+    setTimeout(async () => {
```

Comment:
Why is this necessary now? Also we can replace it with `setImmediate`

Reply:
trying to do the same. I didn't know about the `setImmediate`. Changing it.